s3path.go

```diff
@@ -32,10 +32,15 @@ func urlToS3Path(url *url.URL) (*s3Path, error) {
 		return nil, errNoBucketName
 	}
 
+	path := url.RawPath
+	if path == "" {
+		path = url.Path
+	}
```

Comment:
Could you elaborate on this? Is it correct to use unescaped path in some case?

Reply:
I've added explanation on this bellow.